Index: community-service/src/main/java/com/example/community/service/CommunityService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.community.service;\r\n\r\nimport com.example.community.entity.Community;\r\nimport com.example.community.repository.CommunityRepository;\r\nimport org.springframework.beans.factory.annotation.Autowired;\r\nimport org.springframework.stereotype.Service;\r\n\r\nimport java.util.List;\r\n\r\n@Service\r\npublic class CommunityService {\r\n\r\n    @Autowired\r\n    CommunityRepository communityRepository;\r\n\r\n    public List<Community> findAllCommunities() {\r\n        return communityRepository.findAll();\r\n    }\r\n\r\n    public Community findCommunityById(Long communityId) {\r\n        return communityRepository.findCommunityByCommunityId(communityId);\r\n    }\r\n\r\n    public Community saveCommunity(Community community) {\r\n        return communityRepository.save(community);\r\n    }\r\n\r\n\r\n    public Community deleteCommunity(Long communityId) {\r\n        return communityRepository.deleteCommunityByCommunityId(communityId);\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/community-service/src/main/java/com/example/community/service/CommunityService.java b/community-service/src/main/java/com/example/community/service/CommunityService.java
--- a/community-service/src/main/java/com/example/community/service/CommunityService.java	(revision 2f10d892c151c721aa731bd42492d8f4fb71b5d0)
+++ b/community-service/src/main/java/com/example/community/service/CommunityService.java	(date 1637603946194)
@@ -3,9 +3,12 @@
 import com.example.community.entity.Community;
 import com.example.community.repository.CommunityRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
 
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class CommunityService {
@@ -13,20 +16,38 @@
     @Autowired
     CommunityRepository communityRepository;
 
+    @Autowired
+    private RestTemplate restTemplate;
+
     public List<Community> findAllCommunities() {
+        LogRequest(1L, "GET", "Find all communities");
         return communityRepository.findAll();
     }
 
-    public Community findCommunityById(Long communityId) {
-        return communityRepository.findCommunityByCommunityId(communityId);
+    public Optional<Community> findCommunityById(Long communityId) {
+        LogRequest(1L, "GET", "Find community");
+        return communityRepository.findById(communityId);
     }
 
     public Community saveCommunity(Community community) {
+        LogRequest(1L, "POST", "Save community");
         return communityRepository.save(community);
     }
 
+    public void updateCommunity(Community community) {
+        LogRequest(1L, "PUT", "Update community");
+        communityRepository.save(community);
+    }
 
-    public Community deleteCommunity(Long communityId) {
-        return communityRepository.deleteCommunityByCommunityId(communityId);
+    public void deleteCommunity(Long communityId) {
+        LogRequest(1L, "DELETE", "Delete community");
+        communityRepository.deleteById(communityId);
+    }
+
+    private void LogRequest(Long userId, String action, String description) {
+        Log log = new Log(userId, "Community-service", action, description);
+        HttpEntity<Log> request = new HttpEntity<>(log);
+        restTemplate.postForObject("http://logging-service/logs/saveLog", request, Log.class);
+        System.out.println(log.toString());
     }
 }
